server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java

```diff
@@ -633,17 +644,24 @@ private Map<KeyExtent,TabletMetadata> reserveFilesInner(Collection<KeyExtent> ex
     }
   }
 
-  protected ScanReservation reserveFiles(Collection<KeyExtent> extents)
-      throws NotServingTabletException, AccumuloException {
+  protected ScanReservation reserveFiles(Map<KeyExtent,List<TRange>> extents)
+      throws AccumuloException {
 
     long myReservationId = nextScanReservationId.incrementAndGet();
 
-    Map<KeyExtent,TabletMetadata> tabletsMetadata = reserveFilesInner(extents, myReservationId);
+    Set<KeyExtent> failedReservations = new HashSet<>();
+    Map<KeyExtent,TabletMetadata> tabletsMetadata =
+        reserveFilesInner(extents.keySet(), myReservationId, failedReservations);
     while (tabletsMetadata == null) {
-      tabletsMetadata = reserveFilesInner(extents, myReservationId);
+      tabletsMetadata = reserveFilesInner(extents.keySet(), myReservationId, failedReservations);
     }
+    // Convert failures
```

Comment:
We could add a validation on the return.

```suggestion
    if(!Collections.disjoint(tabletsMetadata.keySet(), failures) || !extents.keySet().equals(Sets.union(tabletsMetadata.keySet(), failures)) {
        throw new AssertionError("bug in reserverFilesInner "+extents.keySet()+" "+tabletsMetadata.keySet()+" "+failures);
    }

    // Convert failures
```

Reply:
Addressed in 381dac3